packages/web-components/src/search-widget/client/utils/__tests__/google-map.ts

```diff
@@ -0,0 +1,47 @@
+import { showPropertiesMarker } from '../google-map'
+import createGoogleMapsMock from '../../../../common/utils/__mocks__/mock-google-map'
+import { property } from '../../../../common/utils/__mocks__/property'
+import { createMarker } from '../../../../common/utils/map-helper'
+import searchWidgetStore, { SearchWidgetStore } from '../../core/store'
+
+jest.mock('../../../../common/utils/map-helper')
+
+describe('google-map utils', () => {
+  it('should return correctly', () => {
+    const maps = createGoogleMapsMock() as any
+    maps.fitBounds = jest.fn()
+    const storeInstance: SearchWidgetStore = {
+      initializers: {
+        apiKey: 'key',
+        theme: {},
+      },
+      properties: { _embedded: [property] },
+      propertyImages: {},
+      selectedProperty: null,
+      markers: [],
+      mapLoading: false,
+    }
+
+    showPropertiesMarker(maps, storeInstance, searchWidgetStore)
+    expect(createMarker).toHaveBeenCalledTimes(1)
+  })
+
+  // it('should return correctly when no properties', () => {
```

Comment:
See my comment about tests - we should test this function better - it is quite complex - recommend as a start point breaking down into blocks and testing them separately. Happy to help with this